controllers/ovndbcluster_controller.go

```diff
@@ -662,7 +665,7 @@ func (r *OVNDBClusterReconciler) reconcileNormal(ctx context.Context, instance *
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.ReadyCount > 0 && len(svcList.Items) > 0 {
+	if statefulset.IsReady(deploy) && len(svcList.Items) > 0 {
```

Comment:
I may be wrong here but I don't really know why we need this additional condition there really. Maybe @karelyatin will know something more about it

Reply:
We need to ensure statefulset is ready before setting Ready Condition

Reply:
thats what `statefulset.IsReady(deploy)` is checking [1]. it is not just checking that we have >0 replica, it is checking:
```
// IsReady - validates when deployment is ready deployed to whats being requested
// - the requested replicas in the spec matches the ReadyReplicas of the status
// - both when the Generatation of the object matches the ObservedGeneration of the Status
```

[1] https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/statefulset/statefulset.go#L142-L145